crates/services/operatorsinfo/src/operatorsinfo_inmemory.rs

```diff
@@ -43,12 +43,19 @@ pub struct OperatorInfoServiceInMemory {
     pub_keys: UnboundedSender<OperatorsInfoMessage>,
 }
 
+/// Source of the operator info.
+#[derive(Debug, Clone, PartialEq)]
+enum StateSource {
+    Historic,
+    Event,
+}
+
 /// State of the operator info service.
 #[derive(Debug, Clone)]
 struct OperatorState {
-    operator_info_data: Arc<RwLock<HashMap<Address, OperatorPubKeys>>>,
-    operator_addr_to_id: Arc<RwLock<HashMap<Address, OperatorId>>>,
-    socket_dict: Arc<RwLock<HashMap<OperatorId, String>>>,
+    operator_info_data: Arc<RwLock<HashMap<Address, (StateSource, OperatorPubKeys)>>>,
+    operator_addr_to_id: Arc<RwLock<HashMap<Address, (StateSource, OperatorId)>>>,
+    socket_dict: Arc<RwLock<HashMap<OperatorId, (StateSource, String)>>>,
```

Comment:
StateSource is used to now if the data we try to insert is more recent or not than the one already store.
Trying to replace a Event data by a History data should be avoided.